### PR TITLE
#162949007 Enable bookmarking of articles

### PR DESCRIPTION
**Pull request by charitymarani**
**What does this PR do?**

1. Adds  endpoints to bookmark ,unbookmark and fetch bokmarked articles

Endpoints involved:
```
PUT  /api/v1/articles/<slug>/bookmark/
GET  /api/v1/articles/bookmarks/
```

2. Adds tests for bookmarking articles

**How to manually test these features**
Clone this Repo
`git checkout ft-bookmark-articles-162949007`
Make Migrations
Run the server
Register users and Add an article
Test the bookmark endpoints on postman
**NB**
The functionality for bookmarking and unbookmarking works like a toggle

**Description of tasks to be completed**
-Users should be able to bookmark articles for reading later
-Users should be to  remove articles from bookmarks
- Users should be able to fetch all their bookmarked articles

**Relevant pivotal tracker stories**
[#162949007](https://www.pivotaltracker.com/story/show/162949007)
**Screenshots**

**Bookmark an article**

screenshot 2019-01-30 at 09 47 13

**Remove article from bookmarks**

screenshot 2019-01-30 at 09 47 30

**Fetch all bookmarked articles for a specific user**

screenshot 2019-01-29 at 17 12 14

**User can't bookmark non-existing article**

screenshot 2019-01-29 at 17 14 02

### CODE REVIEW

**Review by eduhmik — COMMENTED** [review]
Hey @Charity, please do not push to your PR migrations and any files you have not changed. @Kezzy has advised us to only add the specific files we have modified only next time

**Review by Hesbon5600 — COMMENTED** [review]
Hey, I have a suggestion. Instead of just 'Article has been bookmarked!', could the message include the article itself

File: authors/apps/articles/tests/base_test.py [hunk 1]
```diff
@@ -83,6 +83,10 @@ def login_user2(self):
         token = response.data['token']
         return token
 
+    def authorize(self):
```
Comment (Nicanor008) [inline authors/apps/articles/tests/base_test.py]: intelligent way of abstracting user authorization

**Comment by charitymarani** [comment]
> Hey, I have a suggestion. Instead of just 'Article has been bookmarked!', could the message include the article itself

Fixed this @Hesbon5600

**Comment by charitymarani** [comment]
> Hey @charity, please do not push to your PR migrations and any files you have not changed. @kezzy has advised us to only add the specific files we have modified only next time

@eduhmik Fixed this

**Review by Nyakaru — APPROVED** [review]

**Review by kzyangiro — CHANGES_REQUESTED** [review]
@charitymarani Remove profiles migrations.

**Comment by charitymarani** [comment]
> @charitymarani Remove profiles migrations.

@kzyGit I made changes to profile model so the migrations are modified

**Review by kzyangiro — APPROVED** [review]

**Pull request merged**